File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
    var answer = [];
    return strings.sort().sort((a, b) => {
        a = a.substr(n, 1);
        b = b.substr(n, 1);
        console.log("a = " + a + " b = " + b);
        if(a > b) return 1;
        else if(a < b) return -1;
        else return 0;
    });
}

File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;

class Solution {
    public Object[] solution(String[] strings, int n) {
        
        return Arrays.stream(strings)
                    .sorted()
                    .sorted((a, b) -> (a.substring(n, n+1)).compareTo(b.substring(n, n+1)))
                    .toArray();
    }
}